Source/WebKit/WebProcess/WebProcess.cpp

```diff
@@ -266,7 +266,10 @@ using namespace WebCore;
 
 NO_RETURN static void callExit(IPC::Connection*)
 {
-#if OS(WINDOWS)
+#if PLATFORM(GTK) || PLATFORM(WPE)
+    // We use g_error() here to cause a crash and allow debugging this unexpected late exit.
+    g_error("WebProcess didn't exit as expected after the UI process connection was closed");
```

Comment:
The only problem here is the name of this function is "callExit" and now it's calling something else instead. I would do this:

 * Add `#if !PLATFORM(GTK) && !PLATFORM(WPE) guards around `callExit`
 * Call g_error() directly in `callExitSoon` instead of using `callExit`
 * Rename `callExitSoon` to `crashAfter10Seconds`